src/ethereum_test_specs/helpers.py

```diff
@@ -0,0 +1,100 @@
+"""
+Helper functions
+"""
+
+from dataclasses import dataclass
+from typing import Dict, List
+
+from ethereum_clis import Result
+from ethereum_test_exceptions import ExceptionMapper, UndefinedException
+from ethereum_test_types import Transaction
+
+
+@dataclass
+class TransactionExceptionInfo:
+    """Info to print transaction exception error messages"""
+
+    t8n_error_message: str | None
+    transaction_ind: int
+    tx: Transaction
+
+
+def verify_transaction_exception(
+    exception_mapper: ExceptionMapper, info: TransactionExceptionInfo
+):
+    """Verify transaction exception"""
+    exception_info = f"TransactionException (pos={info.transaction_ind}, nonce={info.tx.nonce})\n"
+    expected_error: bool = info.tx.error is not None or (
+        isinstance(info.tx.error, list) and len(info.tx.error) != 0
+    )
+
+    # info.tx.error is expected error code defined in .py test
+    if expected_error and not info.t8n_error_message:
+        raise Exception(f"{exception_info} Error: tx expected to fail succeeded")
```

Comment:
You could define a new Exception class (or classes) to help with these exceptions that handle the relevant info instead of using a `TransactionExceptionInfo` class. I think this would be less surprising to the reader and consistent with how we implement this elsewhere.

For example, this Exception class (see also its base class):
https://github.com/ethereum/execution-spec-tests/blob/5d8d350470f45948c7813fdba21eb8c54a035b67/src/ethereum_test_specs/eof.py#L72-L85

and usage of it here:
https://github.com/ethereum/execution-spec-tests/blob/5d8d350470f45948c7813fdba21eb8c54a035b67/src/ethereum_test_specs/eof.py#L241-L272

Reply:
I think TransactionExceptionInfo  still needed. because it has info from blocks but there is also info from state test

Reply:
done